hls4ml/converters/keras/core.py

```diff
@@ -120,3 +120,19 @@ def parse_batchnorm_layer(keras_layer, input_names, input_shapes, data_reader, c
         layer['n_filt']=input_shapes[0][3]
 
     return layer, [shape for shape in input_shapes[0]]
+
+@keras_handler('Permute')
+def parse_permute_layer(keras_layer, input_names, input_shapes, data_reader, config):
+    assert(keras_layer['class_name'] == 'Permute')
+
+    layer = parse_default_keras_layer(keras_layer, input_names)
+
+    layer['class_name'] = 'Transpose'
+    swap = keras_layer['config']['dims']
+    perm = list(range(len(input_shapes[0])))
+    perm[swap[0]], perm[swap[1]] = perm[swap[1]], perm[swap[0]]
```

Comment:
Hmm... in 3d this breaks. `perm` should just contain values of `dims` with a 0-based index. Aka `perm = [dim - 1 for dim in keras_layer['config']['dims']]`

Reply:
yes, agreed.. i think it can be 

```python
layer['class_name'] = 'Transpose'
perm = [dim - 1 for dim in keras_layer['config']['dims']]
layer['perm'] = perm
output_shape = [input_shapes[0][0]] + [input_shapes[0][p+1] for p in perm]
```

assuming `input_shapes[0][0]` is `None`